web-client/src/presenter/actions/StartCase/setDefaultContactStateAction.ts

```diff
@@ -0,0 +1,53 @@
+import { PARTY_TYPES } from '@shared/business/entities/EntityConstants';
+import { showContactsHelperUpdated } from '@web-client/presenter/computeds/showContactsHelperUpdated';
+import { state } from '@web-client/presenter/app.cerebral';
+
+export const setDefaultContactStateAction = ({
+  applicationContext,
+  get,
+  props,
+  store,
+}: ActionProps<{
+  key: string;
+  value: string;
+}>) => {
+  const partyType = get(state.form.partyType);
+
+  const { showContactSecondary } = showContactsHelperUpdated(
+    partyType,
+    PARTY_TYPES,
+    props,
+  );
+
+  const { COUNTRY_TYPES } = applicationContext.getConstants();
+
+  const defaultContact = { countryType: COUNTRY_TYPES.DOMESTIC };
+
+  const TYPES = [
+    'filingType',
+    'businessType',
+    'otherType',
+    'estateType',
+    'minorIncompetentType',
+  ];
+
+  if (TYPES.includes(props.key)) {
```

Comment:
The early return here is a bit sneaky. Might be worth considering refactoring lines 36-52 to address that and reduce duplicated code.